api/models/user.go

```diff
@@ -112,3 +114,59 @@ func UpdateUserGenerationTimes(userId string, lastGenerated int64, newTimeout in
 
 	return err
 }
+
+func CheckCodeExistence(email string, code string) bool {
+
+	var usercode interfaces.ValidationCode
+	filter := bson.D{{Key: "email", Value: email}}
+	project := bson.D{{Key: "email", Value: 1}, {Key: "validationCode", Value: 1}, {Key: "timeExpiration", Value: 1}}
```

Comment:
Actually, the project isn't needed, you can just get the entire user and it can be safely parsed into `interfaces.ValidationCode`: 

```go
var usercode interfaces.ValidationCode
filter := bson.D{{Key: "email", Value: email}}
err := Collection.FindOne(context.TODO(), filter).Decode(&usercode)
```